From 30c4da4a91a25c3e44b7558d14e3728c2d9aa8cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 13 Aug 2023 21:20:32 +0200
Subject: [PATCH] [Thumbnail] Fix lookup of legacy thumbnailer plugins

For most legacy plugins, "filename" and plugin id are the same, but
the marble plugin "marblethumbnail" provides e.g. marble_thumbnail_kml.

The KIO previewjob.cpp uses the filename:
`job->addMetaData(QStringLiteral("plugin"), currentItem.plugin.fileName());`

Regression from commit 69dc61d7dd34 ("Utilize
KIO::PreviewJob::availableThumbnailerPlugins").


(cherry picked from commit e68a2f24cdd940a126e5b0e8d14285738d32fa14)
---
 thumbnail/thumbnail.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/thumbnail/thumbnail.cpp b/thumbnail/thumbnail.cpp
index 7fef91b5d..c61384798 100644
--- a/thumbnail/thumbnail.cpp
+++ b/thumbnail/thumbnail.cpp
@@ -676,7 +676,7 @@ ThumbCreatorWithMetadata* ThumbnailProtocol::getThumbCreator(const QString& plug
         } else {
             const QVector<KPluginMetaData> plugins = KIO::PreviewJob::availableThumbnailerPlugins();
             auto findPluginIt = std::find_if(plugins.begin(), plugins.end(), [&plugin](const KPluginMetaData &data) {
-                return data.pluginId() == plugin;
+                return data.fileName() == plugin;
             });
             if (findPluginIt != plugins.end()) {
                 data = *findPluginIt;
-- 
GitLab

